torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,467 @@
+            sample_: dict[str, Tensor] = {"image": image, "mask": mask}
+
+            transformed = self.transforms(sample_)
+
+            image = Rearrange("h w c -> c h w")(transformed["image"])
+            mask = Rearrange("h w c -> c h w")(transformed["mask"])
+
+        sample: dict[str, Tensor] = {"image": image, "mask": mask}
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.ids)
+
+    def _load_raster(self, filename: int, source: str) -> Tensor:
+        """Load a single raster image or target.
+
+        Args:
+            filename: name of the file to load
+            source: the filename of the modality
+
+        Returns:
+            the raster image or target
+        """
+        with rasterio.open(os.path.join(self.root, source, f"{filename}.tif")) as f:
+            array = f.read()
+            if array.dtype == np.uint16:
+                array = array.astype(np.int32)
+            tensor = torch.from_numpy(array)
+            return tensor
+
+    def _verify(self, url: str, md5: Optional[str] = None) -> None:
+        """Verify the integrity of the dataset.
+
+        Args:
+            url: url to the file
+            md5: md5 of the file to be verified
+
+        Raises:
+            RuntimeError: if dataset is not found
+        """
+        filepath = os.path.join(self.root, url.split("/")[-1])
+        if os.path.isfile(filepath) and filepath.endswith(".zip"):
+            if self.checksum and not check_integrity(filepath, md5):
+                raise RuntimeError("Dataset found, but corrupted.")
+            self._extract(url)
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                "Dataset not found in `root` directory and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        elif self.download and url.endswith(".csv"):
+            # Download the split file
+            download_url(url, self.root, filename=os.path.split(url)[1], md5=None)
+
+        elif self.download and not url.endswith(".csv"):
+            # Download the dataset
+            self._download(url, md5)
+            self._extract(url)
+
+    def _download(self, url: str, md5: Optional[str]) -> None:
+        """Download the dataset."""
+        download_url(
+            url,
+            self.root,
+            filename=os.path.split(url)[1],
+            md5=md5 if self.checksum else None,
+        )
+
+    def _extract(self, url: str) -> None:
+        """Extract the dataset."""
+        filepath = os.path.join(self.root, os.path.split(url)[1])
+        extract_archive(filepath)
+
+    def _merge_parts(self, root: str, modality: str) -> None:
```

Comment:
I don't really understand what this function does. It just moves both parts into the same directory? Couldn't the dataset be loaded into memory and then joined afterwards without changing the file structure? That would make it easier to reuse without having to re-download the dataset each time

Reply:
By "loading into memory" I just mean maintaining either a list of all file paths or actually loading all files in `__init__`. The former is probably sufficient and more memory efficient. In fact, you already do this by creating `self.ids`. There's nothing in `__getitem__` that requires all files to be in the same directory, right? I guess I just don't think moving all these files around has any benefit and adds a lot of extra code.

Reply:
Will look into this soon but can also do so after the release. Please let me know if you have a preference